agent/backend/pktvisor/pktvisor.go

```diff
@@ -412,11 +412,37 @@ func (p *pktvisorBackend) scrapeOpenTelemetry() (err error) {
 		p.logger.Error("failed to create a receiver", zap.Error(err))
 	}
 
-	err = p.exporter.Start(ctx, nil)
-	if err != nil {
-		p.logger.Error("otel mqtt exporter startup error", zap.Error(err))
-		return err
-	}
+	var errStartExp error
+	go func() {
+		startExpCtx := context.WithValue(ctx, "routine", "startExposrter")
```

Comment:
I believe you will want a cancelFunction here also, this way we will handle things better with cancelling the context after the correct initialization of the Exporters, also there is a typo in `startExposrter`

Reply:
And don' t forget to defer the call so the go func returning would cancel the context.